test/integration/features/basic.feature

```diff
@@ -10,6 +10,61 @@ Feature: Basic
      Then Minishift should have state "Running"
       And Minishift should have a valid IP address
 
+  Scenario: User is able to do ssh into Minishift VM
+    Given Minishift has state "Running"
+     When executing "minishift ssh echo hello"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      hello
+      """
+
+  Scenario: User is able to set custom Docker specific environment variables
+    Given Minishift has state "Running"
+     When executing "minishift ssh cat /var/lib/boot2docker/profile"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      export "FOO=BAR"
+      export "BAZ=BAT"
+      """
+
+  Scenario Outline: User can set, get, view and unset values in configuration file
+    User is able to setup persistent configuration of Minishift using "config" command
+    and its subcommands.
+    Given Minishift has state "Running"
+     When executing "minishift config set <property> <value>"
+     Then JSON config file contains key "<property>" with value "<value>"
```

Comment:
I suggest to make this more generic, something along the lines:

    Then config file 'config/config.json' contains <regexp>

"config file" is a trigger to tell us that we should look for a file in $MINISHIFT_HOME. We then explicitly specify the file we expect to be changed. opposed to saying JSON config. The latter is ambiguous, since we have more than one JSON configuration file. We really want to have a generic method to check that an actual change happened in a given (config) file. Similar reasoning for suggesting the use of 'regex'. It is more generic and can easily be used to match key value pairs as well.